src/OpenVsixSignTool.Core/OpcPackageTimestampBuilder.cs

```diff
@@ -43,58 +52,100 @@ public Task<TimestampResult> SignAsync(Uri timestampServer, HashAlgorithmName ti
             {
                 throw new ArgumentException("The timestamp server must be an absolute URI.", nameof(timestampServer));
             }
-            var oid = HashAlgorithmTranslator.TranslateFromNameToOid(timestampAlgorithm);
             using (var nonce = new TimestampNonceFactory())
             {
-                var parameters = new CRYPT_TIMESTAMP_PARA();
-                parameters.cExtension = 0;
-                parameters.fRequestCerts = true;
-                parameters.Nonce.cbData = nonce.Size;
-                parameters.Nonce.pbData = nonce.Nonce;
-                parameters.pszTSAPolicyId = null;
-                var (signatureDocument, timestampSubject) = GetSignatureToTimestamp(_part);
-                var winResult = Crypt32.CryptRetrieveTimeStamp(
-                    timestampServer.AbsoluteUri,
-                    CryptRetrieveTimeStampRetrievalFlags.NONE,
-                    (uint)Timeout.TotalMilliseconds,
-                    oid.Value,
-                    ref parameters,
-                    timestampSubject,
-                    (uint)timestampSubject.Length,
-                    out var context,
-                    IntPtr.Zero,
-                    IntPtr.Zero
-                );
-                if (!winResult)
+#if NET462 
```

Comment:
This is all super "ew" but working is a start. We can factor this a bit once we are confident it works correctly.